Pargi/MapViewController.swift

```diff
@@ -61,10 +63,23 @@ class MapViewController: UIViewController, MKMapViewDelegate, UIGestureRecognize
             return
         }
         
+        // Stop trackign user location and show button for centering
+        self.locateUserButton.isHidden = false
+        self.trackUserLocation = false
+        
         // Panning has finished, update our visible zones listing
         self.updateVisibleZones()
     }
     
+    @IBAction func locateUser(_ sender: UIButton) {
+        if let currentUserLocation = self.currentUserLocation {
+            self.mapView.setCenter(currentUserLocation.coordinate, animated: false)
+        }
+        self.mapView.camera.heading = 0
```

Comment:
These two changes (the center + the heading) could be done as a single camera change that I think we should probably animate (would give more context as to which direction the user had moved to). Thoughts?

Reply:
I tried doing it first with animation, but it doesn’t work really well. As the map listens to user location change all the time, it can mess up the animation because of the delay and not center the user correctly if the device actually moves while the centering animation is running.